src/org/openbravo/erpCommon/ad_callouts/SL_Order_Amt.java

```diff
@@ -220,61 +223,48 @@ protected void execute(CalloutInfo info) throws ServletException {
       info.addResult("inppricestd", priceStd);
     }
 
-    // Calculate Discount
-    if (StringUtils.equals(strChanged, "inppricelist")
+    BigDecimal price = isTaxIncludedPriceList ? grossPriceList : netPriceList;
+    BigDecimal priceToSubtract = isTaxIncludedPriceList ? grossUnitPrice : priceActual;
+    // Discount calculated with the actual values of the order line
+    BigDecimal calculatedDiscount = price.subtract(priceToSubtract)
```

Comment:
Remember to always compare the constant (BigDecimal.ZERO) first.

Reply:
Done